agent/cache/cache.go

```diff
@@ -234,6 +239,28 @@ func (c *Cache) RegisterType(n string, typ Type) {
 	c.types[n] = typeEntry{Name: n, Type: typ, Opts: &opts}
 }
 
+// ReloadOptions updates the cache with the new options
+// return true if Cache is updated, false if already up to date
+func (c *Cache) ReloadOptions(options Options) bool {
+	options = applyDefaultValuesOnOptions(options)
+	if c.options.EntryFetchRate != options.EntryFetchRate || c.options.EntryFetchMaxBurst != options.EntryFetchMaxBurst {
```

Comment:
I think this function may be slightly easier to read by inverting this condition, and reducing the indentation:

```go
if c.options.EntryFetchRate == options.EntryFetchRate && c.options.EntryFetchMaxBurst == options.EntryFetchMaxBurst {
    return false
}

...
```

Or maybe this could even be moved into a method `func (o Options) Equal() bool`, which would work with libraries like [go-cmp](https://godoc.org/github.com/google/go-cmp/cmp) which we may try to use in the future.

```go
if c.options.Equal(options) {
    return false
}

```

Reply:
done